master/buildbot/worker/base.py

```diff
@@ -678,20 +679,19 @@ def substantiate(self, sb, build):
             self._clearBuildWaitTimer()
             self._setBuildWaitTimer()
             return defer.succeed(True)
-        if self.substantiation_deferred is None:
+        if not self._substantiation_notifier:
             if self.parent and not self.missing_timer:
                 # start timer.  if timer times out, fail deferred
                 self.missing_timer = self.master.reactor.callLater(
                     self.missing_timeout,
                     self._substantiation_failed, defer.TimeoutError())
-            self.substantiation_deferred = defer.Deferred()
             self.substantiation_build = build
             if self.conn is None:
                 d = self._substantiate(build)  # start up instance
                 d.addErrback(log.err, "while substantiating")
```

Comment:
@tomprince 

if substanciate errbacks, the notifier will never be called. Is this on purpose?